### PR TITLE
Plots and embeddings documentation

### PR DESCRIPTION
**Pull request by brimoor**
Docs updates
- Adds a user guide page detailing the new plotting functionality
- Adds a section on embedding visualization to the brain user guide page
- Adds ~30% of an embeddings tutorial

Other changes
- Interactive plots will now respect selected samples/labels when no view is loaded (ie full dataset)
- Fixed a bug that prevented interactive confusion matrices from working when COCO-style detection evaluation is run with no `eval_key`
- Allows the initial size of a `Session` object to be set via a new `launch_app(..., height=)` parameter
- Renames `PlotlyHeatmap` to `InteractiveHeatmap`, for consistency
- Prevents an error from being raised when (trying to) freeze matplotlib figures when using the `%matplotlib widget` backend

### CODE REVIEW

File: docs/source/user_guide/brain.rst [hunk 1]
```diff
@@ -20,35 +20,204 @@ science.
...
-* **Uniqueness**: During the training loop for a model, the best results will
+* :ref:`Visualizing embeddings <brain-embeddings-visualization>`:
+  Tired of combing through individual images/videos
+  and staring at aggregate performance metrics trying to figure out how to
+  improve the performance of your model? Visualizing your dataset in a
+  low-dimensional embedding space reveal patterns and clusters in your data
+  that can help you answer many important questions about your data, from
+  identifying the most critical failure modes of your model, to isolating
+  examples of critical scenarios, to recommending new samples to add to your
+  training dataset, and more! The FiftyOne Brain provies a poweful
```
Comment (ehofesmann) [inline docs/source/user_guide/brain.rst:31]: "powerful"

File: docs/source/user_guide/brain.rst [hunk 2]
```diff
@@ -20,35 +20,204 @@ science.
...
+Visualizing embeddings
+______________________
+
+The FiftyOne Brain provides a powerful
+:meth:`compute_visualization() <fiftyone.brain.compute_visualization>` method
+that you can use to generate low-dimensional representations of the samples
+and/or individual objects in your datasets.
+
+These representations can be visualized via
+:ref:`interactive plots <interactive-plots>`, which can be connected to the
```
Comment (ehofesmann) [inline docs/source/user_guide/brain.rst:80]: Add a note that these are currently only interactive in Jupyter notebooks.

File: docs/source/user_guide/plots.rst [hunk 3]
```diff
@@ -0,0 +1,969 @@
...
+data that you can use to identify data clusters corresponding to model failure
+modes, annotation gaps, and more.
+
+What do we mean by **interactive plots**? First, FiftyOne plots are
+`powered by Plotly <https://plotly.com/python>`_, which means they are
+responsive JavaScript-based plots that can be zommed, panned, and lasso-ed.
+Second, FiftyOne plots can be linked to the :ref:`FiftyOne App <fiftyone-app>`,
+so that selecting points in a plot will automatically load the corresponding
+samples/labels in the App (and vice versa) for you to visualize! Linking plots
+to their source media is a powerful paradigm that it is a critical part of any
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:27]: Used powerful a lot, recommend changing this to:
"Linking plots to their source media is a compelling paradigm that is a critical part..."

File: docs/source/user_guide/plots.rst [hunk 4]
```diff
@@ -0,0 +1,969 @@
...
+    meantime, you can still use FiftyOne's plotting features in other
+    environments, but you must manually call
+    :meth:`plot.show() <fiftyone.core.plots.base.Plot.show>` to update the
+    state of a plot to match the state of a connected |Session|, and any
+    callbacks that would normally be triggered in response to interacting with
+    a plot will not be triggered.
+
+    See :ref:`this section <working-in-notebooks>` for more information.
+
+Thw two main classes of |ResponsivePlot| are explained next.
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:72]: The

File: docs/source/user_guide/plots.rst [hunk 5]
```diff
@@ -0,0 +1,969 @@
...
+:ref:`interactive heatmaps <confusion-matrices>`.
+
+.. image:: ../images/plots/mnist-interactive1.gif
+   :alt: interactive-plots
+   :align: center
+
+View plots
+----------
+
+|ViewPlot| is a class of plots whose state is automatically updated whenever
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:97]: Specifically state that they are not bidirectional. Also mention examples of types of `ViewPlots` here like for interactive plots.

File: docs/source/user_guide/plots.rst [hunk 6]
```diff
@@ -0,0 +1,969 @@
...
+
+If you wish to use the ``matplotlib`` backend for any interactive plots, refer
+to :ref:`this section <matplotlib-in-notebooks>` for setup instructions.
+
+.. _embeddings-plots:
+
+Visualizing embeddings
+______________________
+
+The :ref:`FiftyOne Brain <fiftyone-brain>` provides a powerful
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:179]: Powerful again, maybe change to "valuable" or "convenient"

File: docs/source/user_guide/plots.rst [hunk 7]
```diff
@@ -0,0 +1,969 @@
...
+As you can see, the 2D embeddings are naturally clustered according to their
+ground truth label!
+
+Interactive plots
+-----------------
+
+The real power of
+:meth:`compute_visualization() <fiftyone.brain.compute_visualization>` comes
+when you associate the scatterpoints with the samples in a |Dataset| and then
+attach it to a |Session|.|Dataset| and then attach it to a |Session|.
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:246]: Formatting issue:

image

File: docs/source/user_guide/plots.rst [hunk 8]
```diff
@@ -0,0 +1,969 @@
...
+:ref:`attached to the App <attaching-plots>`.
+
+In this setup, the scatterplot renders each sample using its corresponding 2D
+embedding generated by
+:meth:`compute_visualization() <fiftyone.brain.compute_visualization>`, colored
+by the sample's ground truth label.
+
+Since the ``labels`` argument to
+:meth:`results.visualize() <fiftyone.brain.visualization.VisualizationResults.visualize>`
+is categorical, each class is rendered as its own trace and you can click or on
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:259]: Remove "or"

File: docs/source/user_guide/plots.rst [hunk 9]
```diff
@@ -0,0 +1,969 @@
...
+the legend entires to show/hide individual classes, or double-click to
+show/hide all other classes.
+
+When points are lasso-ed in the plot, the corresponding
+samples are automatically selected in the session's current
+:meth:`view <fiftyone.core.session.Session.view>`. Likewise, whenever you
+modify the Session's view, either in the App or by programmatically setting
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:266]: lowercase "session's" for consistency

File: docs/source/user_guide/plots.rst [hunk 10]
```diff
@@ -0,0 +1,969 @@
...
+interactively explore specific cases of your model's performance.
+
+.. note::
+
+    See :ref:`this page <evaluating-models>` for an in-depth guide to using
+    FiftyOne to evaluate models.
+
+The example below demonstrates setting up an interactive confusion matrix for
+the results of evaluating the predictions in the ``predictions`` field of the
+:ref:`quickstart <dataset-zoo-quickstart>` dataset.
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:495]: Phrasing:
"setting up an interactive confusion matrix for the results of an evaluation on the ``predictions`` field of the ..."

File: docs/source/user_guide/plots.rst [hunk 11]
```diff
@@ -0,0 +1,969 @@
...
+   :alt: view-plots
+   :align: center
+
+.. _attaching-plots:
+
+Attaching plots to the App
+__________________________
+
+All |Session| instances provide a
+:meth:`plots attribute <fiftyone.core.session.Session.plots>` attribute that
```
Comment (ehofesmann) [inline docs/source/user_guide/plots.rst:608]: remove "attribute"

**Review by benjaminpkane — APPROVED** [review]
LGTM.

I noticed a couple typos in the embeddings tutorial, but that is still WIP.

**Pull request merged**